app/knight.py

```diff
@@ -0,0 +1,30 @@
+class Knight:
+    def __init__(self, knight: dict) -> None:
+        self.name = knight["name"]
+        self.power = knight["power"]
+        self.hp = knight["hp"]
+        self.armour = knight["armour"]
+        self.weapon = knight["weapon"]
+        self.potion = knight["potion"]
+        self.protection = 0
+
+        # a knight prepares for battle
+        self.battle_preparations()
```

Comment:
not sure it should be in an init method, more logical that knight should prepare for a battle when he is going to a battle, not when he just initialized)